### ISSUE TITLE
MockServer 5.3.0 fails when using Jackson databind 2.9.4 when matching a header name and regex value

### ISSUE THREAD

**Issue by plumpop-admin** [jamesdbloom/mockserver#440]
I am trying to upgrade the Jackson databind dependency (com.fasterxml.jackson.core:jackson-databind) to 2.9.4 to work with MockServer 5.3.0.

When I try to use the sample code in the documentation for matching a request by header name and value regex with Jackson 2.9.4, a com.fasterxml.jackson.databind.exc.MismatchedInputException exception is thrown while attempting to create the headers expectation. At that point, the MockServer is unresponsive.

Note: This does not happen with Jackson databind 2.9.3.

Reproduction Code:
`
new MockServerClient("localhost", 1080)
                .when(
                        request()
                                .withPath("/some/path")
                                .withHeader(
                                        header("Accept.*", ".*gzip.*")
                                )
                )
                .respond(
                        response()
                                .withBody("some_response_body")
                );
`

Stacktrace:
[mock_server_5_3_0_jackson_2_9_4_headers_expectation_stacktrace.txt](https://github.com/jamesdbloom/mockserver/files/1755882/mock_server_5_3_0_jackson_2_9_4_headers_expectation_stacktrace.txt)

**Comment by aheritier** [comment]
Hi @plumpop-admin 

  I was also trying to identify the same issue. In my case, I'm facing it also with an upgrade of Jackson to version 2.9.4 (version defined by Spring-Boot 2.0.0) but with a simple header without any regexp like

```
                                    .withHeaders(
                                            new Header("Content-Type", "application/json; charset=utf-8")
                                    )
```

I confirm that downgrading jackson to 2.9.3 solves the issue. Thanks for the tip.

**Comment by dstarcev** [comment]
For me headers doesn't work at all for both request and response parts with jackson 2.9.4.

at `headers` node it fails with `Cannot deserialize instance of `java.lang.String` out of FIELD_NAME token`

**Comment by ppsnumi** [comment]
throw same exception 

env:
mock-server 5.3.0
jackson 2.9.6

**Comment by chnardi** [comment]
Same problem. I think it's fixed in this commit https://github.com/jamesdbloom/mockserver/commit/d267a2d5bc7f1b48abdf7ade0457244c7761d595 but unfortunately that's only in 5.4.1 which changes several packages and classes, so the migration doesn't seem straightforward

**Comment by aheritier** [comment]
I confirm it seems to be ok with 5.4.1

**Comment by jamesdbloom** [comment]
this issue is now resolved in 5.4.1

**Comment by liulihai0840** [comment]
I met with the same issue.  I will check my mock-server version .
